### DISCUSSION TITLE
Modeling Vector Processor Behavior in gem5

### DISCUSSION THREAD

**Discussion by sheikhfaizanqureshi** [gem5/gem5#1484] [Q&A]
I am studying RVV1.0 by simulating basic kernels in gem5 and exploring the vector instruction flow using the O3PipelineViewer. I configured the vector length (vlen) to 512 bits and the element length (elen) to 64 bits. Based on my understanding, this should mean that in a vector processor model, a simple vadd.vv instruction would take 512/64 = 8 cycles to complete in the execution pipeline (during the issue/complete stage). However, I observed that the vector instructions remain in the execution stage for only one cycle.

This behavior seems more like that of a SIMD processor (1 cycle per vector instruction) than a vector processor, which should take N cycles (where N = vlen/datapath width).

My question is: How should I properly model a vector processor in gem5 to reflect this expected multi-cycle behavior for vector instructions?

**Comment by qqli579** [comment]
Maybe you misunderstood, RVV's vector instructions is SIMD. If the execution unit of Vector is 512 bits, then the vadd.vv command only needs one cycle to be executed, and if the execution unit is 128 bits, then it needs 2 cycles.

**Reply by sheikhfaizanqureshi** [reply]
Thanks for the response. Can you tell me where the width of the SIMD execution unit is configured in the gem5 RVV setup?
